astrality/context.py

```diff
@@ -187,3 +200,14 @@ def update(self, other: Union['Context', dict]) -> None:
         """Overwrite all items from other onto the Context object."""
         for key, value in other.items():
             self.__setitem__(key, value)
+
+    def copy(self) -> 'Context':
+        """Return shallow copy of context."""
+        return Context(self._dict.copy())
+
+    def reverse_update(self, other: Union['Context', dict]) -> None:
+        """Update context while preserving conflicting keys."""
+        original_copy = self._dict.copy()
```

Comment:
don't need to create a copy of the original just for a one-time use as an argument to update since its values will only be read and not updated.

Reply:
That's true! Changing that part.